File: wrappers/java/library/src/com/worldpay/innovation/types/ServiceDeliveryToken.java
package com.worldpay.innovation.types;

import com.worldpay.innovation.wpwithin.rpc.types.ServiceDetails;

/**
 * Created by conor on 29/07/2016.
 */
public class ServiceDeliveryToken {

    public String key;
    public String issued;
    public String expiry;
    public boolean refundOnExpiry;
    public byte[] signature;

    public ServiceDeliveryToken(String key, String issued, String expiry, boolean refundOnExpiry, byte[] signature) {

        this.key = key;
        this.issued = issued;
        this.expiry = expiry;
        this.refundOnExpiry = refundOnExpiry;
        this.signature = signature;
    }

    public String getKey() {
        return key;
    }

    public String getIssued() {
        return issued;
    }

    public String getExpiry() {
        return expiry;
    }

    public boolean isRefundOnExpiry() {
        return refundOnExpiry;
    }

    public byte[] getSignature() {
        return signature;
    }
}

File: wrappers/dotnet/Worldpay.Within/Worldpay.Within/ServiceDeliveryToken.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Worldpay.Innovation.WPWithin
{
    public class ServiceDeliveryToken
    {
        public string Key { get; set; }

        public string Issued { get; set; }

        public string Expiry { get; set; }

        public bool? RefundOnExpiry { get; set; }

        public byte[] Signature { get; set; }

        public override bool Equals(object that)
        {
            return new EqualsBuilder<ServiceDeliveryToken>(this, that)
                .With(m => m.Key)
                .With(m => m.Issued)
                .With(m => m.Expiry)
                .With(m => m.RefundOnExpiry)
                .With(m => m.Signature).Equals();
        }

        public override int GetHashCode()
        {
            return new HashCodeBuilder<ServiceDeliveryToken>(this)
                .With(m => m.Key)
                .With(m => m.Issued)
                .With(m => m.Expiry)
                .With(m => m.RefundOnExpiry)
                .With(m => m.Signature)
                .HashCode;
        }

        public override string ToString()
        {
            return new ToStringBuilder<ServiceDeliveryToken>(this)
                .Append(m => m.Key)
                .Append(m => m.Issued)
                .Append(m => m.Expiry)
                .Append(m => m.RefundOnExpiry)
                .Append(m => m.Signature)
                .ToString();
        }
    }
}
